### PR TITLE
Bump phpstan to level 6

### PR DESCRIPTION
**Pull request by VincentLanglet**
Full bump of phpstan to level 6.

Do you think the `phpstan-type Group` and `Item` is better in the Pool or in the `GroupMenuProvider` ?
I think it could be better in the second one, wdyt @sonata-project/contributors ?

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Adding
```
    /**
     * @phpstan-param class-string<T> $class
     */
```
to the constructor of Admin classes is trggering a lot of new issues.

File: src/Admin/AbstractAdminExtension.php [hunk 1]
```diff
@@ -127,11 +127,6 @@ public function configureDashboardActions(AdminInterface $admin, array $actions)
         return $actions;
     }
 
-    public function configureDashboardButtons(AdminInterface $admin, array $actions): array
-    {
-        return $actions;
-    }
```
Comment (wbloszyk) [inline src/Admin/AbstractAdminExtension.php:133]: Can we remove it without upgrade note? Or is it done in other place?

File: tests/Command/GenerateObjectAclCommandTest.php [hunk 2]
```diff
@@ -112,6 +115,7 @@ public function testExecuteWithManipulatorNotObjectAclManipulatorInterface(): vo
             'sonata.admin.manipulator.acl.object.bar' => new \stdClass(),
         ];
 
+        // @phpstan-ignore-next-line
         $command = new GenerateObjectAclCommand($pool, $aclObjectManipulators);
```
Comment (wbloszyk) [inline tests/Command/GenerateObjectAclCommandTest.php:119]: Will be nice to resolve it in other way but we can accept it.

File: src/Admin/AbstractAdminExtension.php [hunk 3]
```diff
@@ -127,11 +127,6 @@ public function configureDashboardActions(AdminInterface $admin, array $actions)
...
-
```
Comment (wbloszyk) [inline src/Admin/AbstractAdminExtension.php:134]: Can we remove it without upgrade note? Or is it done in other place?

**Review by wbloszyk — APPROVED** [review]
IMO better places them in a class where they are often used. Anyway both solution are good.

File: tests/Command/GenerateObjectAclCommandTest.php (hunk 2 above)
Comment (VincentLanglet) [reply tests/Command/GenerateObjectAclCommandTest.php:119]: When we want to pass a wrong argument and see we're having an exception, I don't see another way

File: src/Admin/AbstractAdminExtension.php (hunk 3 above)
Comment (VincentLanglet) [reply src/Admin/AbstractAdminExtension.php:134]: This was deprecated from the interface in 3.x

**Review by wbloszyk — APPROVED** [review]

**Pull request merged**